### DISCUSSION TITLE
Automaticly close a disclosure when a new one is opened?

### DISCUSSION THREAD

**Discussion by Notaduck** [tailwindlabs/headlessui#897] [Help]
Hi There.
Has anybody successfully implemented a list of disclosures in react, where there only can be one opened disclosure, and the opened will automatically be closed when a new one is opened?

**Comment by mkrkelj1** [comment]
Currently working on this, though my solution is kind of hack-y, as there's no programmatic interface to do this cleanly.

**Reply by Notaduck** [reply]
I would love to see your solution to this!

**Reply by sontiO** [reply]
Did you end up finding a workaround for this?

**Reply by cdt-eth** [reply]
checking in on this! @mkrkelj1

**Reply by raymondkingston** [reply]
Hi @mkrkelj1, I know it's a year later, but would you care to share your solution? Can't believe this option isn't built in by default.
Thanks!

**Comment by IamLizu** [comment]
Here's what I did as a workaround (not necessarily a good way),

In the `Disclosure.Button` or the element that is being clicked, I added the following function as event handler,

```js
const handleDiscloserChange = (idx) => {
    openIndex &&
         openIndex !== idx &&
         document.getElementById(openIndex).click();

    idx === openIndex ? setOpenIndex(null) : setOpenIndex(idx);
};
```

I have a state for the `openIndex` and I have set the `idx` as id in the `Disclosure.Button`. I check whether the there's a `openIndex` and if `openIndex` is similar to the current `idx`, if not, I click the `openIndex` manually to close it. 

Please do suggest improvements and better alternative.
